src/interface_adapters/signup/SignupController.java

```diff
@@ -10,9 +10,9 @@ public SignupController(SignupInputBoundary userSignupUseCaseInteractor) {
         this.userSignupUseCaseInteractor = userSignupUseCaseInteractor;
     }
 
-    public void execute(String username, String password1, String password2) {
+    public void execute(String username, String password1, String password2, boolean isReturning) {
         SignupInputData signupInputData = new SignupInputData(
-                username, password1, password2);
+                username, password1, password2, isReturning);
 
         userSignupUseCaseInteractor.execute(signupInputData);
```

Comment:
I initially added another "execute method, but I felt like a single execute method reflect CA more closely, so instead I added a boolean checker to the InputData which showed whether the user was new or returning, and either switching to login view or invoking the signup process accordingly.